etl/publish.py

```diff
@@ -25,53 +25,60 @@ class CannotPublish(Exception):
 
 
 @click.command()
-@click.option("--dry-run", is_flag=True)
-def publish(dry_run: bool = False) -> None:
+@click.option("--dry-run", is_flag=True, default=False)
+@click.option("--private", is_flag=True, default=False)
+@click.option("--bucket", type=str, help="Bucket name", default=config.S3_BUCKET)
+def publish(dry_run: bool, private: bool, bucket: str) -> None:
     """
     Publish the generated data catalog to S3.
     """
-    sanity_checks()
-    sync_catalog_to_s3(dry_run=dry_run)
+    catalog = Path(DATA_DIR)
+    if not dry_run:
+        assert (
+            private
+        ), "You cannot publish public catalog yet, only private catalogs with flag --private are supported"
+    sanity_checks(catalog)
+    sync_catalog_to_s3(bucket, catalog, dry_run=dry_run)
 
 
-def sanity_checks() -> None:
-    if not (DATA_DIR / "catalog.feather").exists():
+def sanity_checks(catalog: Path) -> None:
+    if not (catalog / "catalog.feather").exists():
         print(
             "ERROR: catalog has not been indexed, refusing to publish", file=sys.stderr
         )
         sys.exit(1)
 
 
-def sync_catalog_to_s3(dry_run: bool = False) -> None:
+def sync_catalog_to_s3(bucket: str, catalog: Path, dry_run: bool = False) -> None:
     s3 = connect_s3()
-    if is_catalog_up_to_date(s3):
+    if is_catalog_up_to_date(s3, bucket, catalog):
         print("Catalog is up to date!")
         return
 
-    sync_datasets(s3, dry_run=dry_run)
+    sync_datasets(s3, bucket, catalog, dry_run=dry_run)
     if not dry_run:
-        update_catalog(s3)
+        update_catalog(s3, bucket, catalog)
 
 
-def is_catalog_up_to_date(s3: Any) -> bool:
-    remote = get_remote_checksum(s3, "catalog.feather")
-    local = files.checksum_file(DATA_DIR / "catalog.feather")
+def is_catalog_up_to_date(s3: Any, bucket: str, catalog: Path) -> bool:
```

Comment:
Suggesting a comment that would help future readers

```suggestion
def is_catalog_up_to_date(s3: Any, bucket: str, catalog: Path) -> bool:
    # The catalog file is synced last -- if it is the same as our local one, then all the remote
    # files will be the same as our local ones too.
```